allennlp/data/dataset_readers/dataset_utils/ontonotes.py

```diff
@@ -0,0 +1,418 @@
+
+            self._process_span_annotations_for_word(conll_components[10:-1],
+                                                    span_labels,
+                                                    current_span_labels)
+
+            # If any annotation contains this word as a verb predicate,
+            # we need to record its index. This also has the side effect
+            # of ordering the verbal predicates by their location in the
+            # sentence, automatically aligning them with the annotations.
+            word_is_verbal_predicate = any(["(V" in x for x in conll_components[11:-1]])
+            if word_is_verbal_predicate:
+                verbal_predicates.append(word)
+
+            self._process_coref_span_annotations_for_word(conll_components[-1],
+                                                          index,
+                                                          clusters,
+                                                          coref_stacks)
+            coreference.append(conll_components[-1])
+
+            sentence.append(word)
+            pos_tags.append(pos_tag)
+            parse_pieces.append(parse_piece)
+            predicate_lemmas.append(lemmatised_word if lemmatised_word != "-" else None)
+            predicate_framenet_ids.append(framenet_id if framenet_id != "-" else None)
+            word_senses.append(int(word_sense) if word_sense != "-" else None)
+            speakers.append(speaker if speaker != "-" else None)
+
+        named_entities = span_labels[0]
+        srl_frames = {predicate: labels for predicate, labels
+                      in zip(verbal_predicates, span_labels[1:])}
+
+        parse_tree = Tree.fromstring("".join(parse_pieces))
+        coref_span_tuples: Set[TypedSpan] = {(cluster_id, span)
+                                             for cluster_id, span_list in clusters.items()
+                                             for span in span_list}
+        return OntonotesSentence(document_id,
+                                 sentence_id,
+                                 sentence,
+                                 pos_tags,
+                                 parse_tree,
+                                 predicate_lemmas,
+                                 predicate_framenet_ids,
+                                 word_senses,
+                                 speakers,
+                                 named_entities,
+                                 srl_frames,
+                                 coref_span_tuples)
+
+    @staticmethod
+    def _process_coref_span_annotations_for_word(label: str,
+                                                 word_index: int,
+                                                 clusters: DefaultDict[int, List[Tuple[int, int]]],
+                                                 coref_stacks: DefaultDict[int, List[int]]) -> None:
+        """
+        For a given coref label, add it to a currently open span/s, complete a span/s or
```

Comment:
I've more commonly seen "span(s)" instead of "span/s" - took me a minute to parse what you meant by "span/s".